### ISSUE TITLE
Unexpected merge commits in "main" branch

### ISSUE THREAD

**Issue by andrewtbiehl** [andrewtbiehl/gaslines#43]
The changes from PRs #40, #41, and #42 were all merged into the `main` branch via the "standard" (i.e. two parent) merge strategy, instead of via the repository's preferred squash merge strategy. These merge commits are now unfortunately a part of the `main` branch commit history so, as a consequence, the `main` branch no longer has an entirely linear history.

This Issue captures the decision of how to best remedy the merge commits that are already present, as well as the determination of how to better avoid this occurrence going forward.

**Comment by andrewtbiehl** [comment]
Upon investigating this issue further, I was able to determine that these commits most likely resulted from an issue with the GitHub UI. For a brief period, clicking the "Squash and merge" button on a PR into a branch with the linear history requirement enabled would actually cause an attempt at creating a standard merge commit. Moreover, if administrative privileges were used to override branch protection rules, this merge commit attempt would be successful. [This screen recording](https://youtu.be/TniJq7by8Zs) demonstrates this issue occurring in a now deleted example repository.

I opened ticket #907411 with [GitHub Support](https://support.github.com/) to notify them of the issue and to hopefully get some information in return. It is worth noting that their responses were very prompt and supportive, which I appreciate a lot. Here are the noteworthy excerpts from their responses:

> It appears that this was related to a change on our side. Our engineers have rolled out a fix for this and you should not run into this in the future.

> I don't believe we have published any specifics around this incident, likely because it affected only a small number of PRs and was reverted quickly.

In other words, GitHub Support was able to confirm that this was an error on the side of GitHub, so my own culpability with regards to these accidental merge commits is minimal, and I don't need to worry too much about it happening again.

**Comment by andrewtbiehl** [comment]
Even though the `main` branch no longer has a linear history, I've decided not to make any modifications to the history to resolve this. Instead, the accidental merge commits will remain as they were committed.

My reasoning behind this decision mainly rests on the fact that, even though there are various methods available for editing the commit history in particular, not all of the changes made to this repository can be amended. Most notably, affected pull requests cannot be overwritten and related reports sent to coveralls cannot be withdrawn. Hence it makes the most sense to me to just leave the history as it is. That being said, I am still working on solutions that will prevent issues like this from occurring again.

**Comment by andrewtbiehl** [comment]
As described in PR #45, the solution I decided upon to prevent issues like this from happening again takes the form of a new GitHub Actions workflow. Please see PR #45 for more information.
